diff -ru src/main/java/com/megacrit/cardcrawl/map/MapRoomNode.java src/main/java/com/megacrit/cardcrawl/map/MapRoomNode.java
--- src/main/java/com/megacrit/cardcrawl/map/MapRoomNode.java	2018-02-08 20:13:24.017210200 -0800
+++ src/main/java/com/megacrit/cardcrawl/map/MapRoomNode.java	2018-02-08 20:25:19.351777300 -0800
@@ -31,6 +31,9 @@
 import java.util.ArrayList;
 import java.util.Iterator;
 
+// ColoredMap imports
+import coloredmap.ColoredMap;
+
 public class MapRoomNode {
     private static final int IMG_WIDTH = (int)(Settings.scale * 64.0f);
     private static final float OFFSET_X = 560.0f * Settings.scale;
@@ -166,7 +169,7 @@
                 } else {
                     AbstractDungeon.getCurrMapNode().taken = true;
                 }
-                Iterator<MapEdge> connectedEdge = AbstractDungeon.getCurrMapNode().getEdgeConnectedTo(this);
+                MapEdge connectedEdge = AbstractDungeon.getCurrMapNode().getEdgeConnectedTo(this);
                 if (connectedEdge != null) {
                     connectedEdge.markAsTaken();
                 }
@@ -321,17 +324,15 @@
         for (MapEdge edge : this.edges) {
             edge.render(sb);
         }
-        if (this.highlighted) {
-            sb.setColor(Color.LIGHT_GRAY);
-        } else {
-            sb.setColor(OUTLINE_COLOR);
-        }
+        
+        // ColoredMap modification
+        ColoredMap.setIconOutlineColor(room, sb);
+        
         sb.draw(this.room.getMapImgOutline(), (float)this.x * SPACING_X + OFFSET_X - 64.0f + this.offsetX, (float)this.y * Settings.MAP_DST_Y + OFFSET_Y + DungeonMapScreen.offsetY - 64.0f + this.offsetY, 64.0f, 64.0f, 128.0f, 128.0f, this.scale * Settings.scale, this.scale * Settings.scale, 0.0f, 0, 0, 128, 128, false, false);
-        if (this.taken) {
-            sb.setColor(AVAILABLE_COLOR);
-        } else {
-            sb.setColor(this.color);
-        }
+        
+        // ColoredMap modification
+        ColoredMap.setIconColor(room, sb);
+        
         sb.draw(this.room.getMapImg(), (float)this.x * SPACING_X + OFFSET_X - 64.0f + this.offsetX, (float)this.y * Settings.MAP_DST_Y + OFFSET_Y + DungeonMapScreen.offsetY - 64.0f + this.offsetY, 64.0f, 64.0f, 128.0f, 128.0f, this.scale * Settings.scale, this.scale * Settings.scale, 0.0f, 0, 0, 128, 128, false, false);
         if (this.taken || AbstractDungeon.firstRoomChosen && this.equals(AbstractDungeon.getCurrMapNode())) {
             sb.setColor(AVAILABLE_COLOR);
